### ISSUE TITLE
how to specify configuration options for agent

### ISSUE THREAD

**Issue by michaelandsylas234** [MicrosoftDocs/visualstudio-docs#5673]
The page says:
Base/Low/High configurations refer to the sampling rate. For example, Low is 100 samples/second and High is 4000 samples/second. For the VSDiagnostics.exe tool to work with a collection agent, it requires both a DLL and a COM CLSID for the appropriate agent. **The agent might also have additional configuration options.** If you use an agent without a configuration file, use the format in the following command....

However, the form of the "config" string is not given

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e998101e-f5d8-517a-7f74-356b973a2b07
* Version Independent ID: 36f07905-43eb-019a-a0f6-a732b9d10da6
* Content: [Run profiling tools with or without the debugger - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/profiling/running-profiling-tools-with-or-without-the-debugger?view=vs-2019)
* Content Source: [docs/profiling/running-profiling-tools-with-or-without-the-debugger.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/profiling/running-profiling-tools-with-or-without-the-debugger.md)
* Product: **visual-studio-windows**
* Technology: **vs-ide-debug**
* GitHub Login: @Mikejo5000
* Microsoft Alias: **mikejo**

**Comment by Mikejo5000** [comment]
@michaelandsylas234 I'll ask about that, thanks.

**Comment by Mikejo5000** [comment]
Sorry for getting back to this so slowly. I need to update this doc.
Here's the info:

[the form of the config string] is the same info that is in the config file. For the CPU Usage tool in the base format it should be:

VSDiagnostics.exe start <id> /attach:<pid> /loadAgent:4EA90761-2248-496C-B854-3C0399A591A4;DiagnosticsHub.CpuAgent.dll

The config is then any options that are in the config file formatted as properly escaped JSON. Since this is a huge pain from the command line I would highly suggest they use the config files, that is why we added them. Also, this tool was originally added so that our team had a way of controlling the collector outside of VS then we adapted it as a command line scenario for some enterprise customers. This legacy is definitely showing here now that we are trying to document and make it the official way 😊.
